optimizely/decision/composite_feature_service.go

```diff
@@ -47,12 +53,24 @@ func (f CompositeFeatureService) GetDecision(decisionContext FeatureDecisionCont
 		}
 
 		experimentDecision, err := f.experimentDecisionService.GetDecision(experimentDecisionContext, userContext)
-		featureDecision := FeatureDecision{
-			Experiment: experiment,
-			Decision:   experimentDecision.Decision,
-			Variation:  experimentDecision.Variation,
+		// only return the decision if we have a valid variation
+		// @TODO: log the decision made here as debug
+		if experimentDecision.Variation.ID != "" {
```

Comment:
Nit: is it weird that `ID` could be an empty string? This seems like invalid data. How does this end up being an empty string and is there a better representation we can use?

Reply:
Empty string is the zero-value for strings and it is what the variation is initialized with. I toyed with using `nil` for the variation but for that we would need to pass references to the Variation instead of the copy of the Variation because of how it works in Go. Making the function return the reference to the Variation would make it seem like we are returning a mutable struct, which we are not.

Reply:
Shouldn't we confirm that there is no `err` before doing this check?